torch/serialization.py

```diff
@@ -107,6 +107,9 @@ def validate_cuda_device(location):
         location = str(location)
     if not isinstance(location, _string_classes):
         raise ValueError("location should be a string or torch.device")
+    if len(location) > 4 and location[4] != ':':
```

Comment:
I have a lot of questions about this function:
1. This is so adhoc. Why is the ordinary parser used, i.e., by calling `torch.device(location)`? Or better yet, just call this function: https://github.com/pytorch/pytorch/blob/e95dc9814eaf40b5e85020a8a8831d1df187597c/torch/cuda/_utils.py#L5
2. Why is the default device `0` rather than the current default device?

Reply:
This fix is meant for raising an error when the user specifies `cuda0` instead of `cuda:0` function. I just modified the `validate_cuda_device` method that is called by `load` for checking whether the `:` exists or not.

Regarding your queries:
1. I wasn't aware of this function. Using it does not require use of the `if` statement. Sending a fix now.
2. This was there before I submitted this PR and assumed that `device=0` is the default feature intended for this code.

Reply:
Thanks! I think we probably should set `optional=True` for this.

Reply:
Its done. 

CC: @ezyang